shared/src/test/scala/SemaphoreBehavior.scala

```diff
@@ -46,22 +47,24 @@ class SemaphoreBehavior extends munit.FunSuite {
   }
 
   test("no release high-numbered semaphore") {
-    val futs = Async.blocking:
-      val sem = Semaphore(100)
-      val count = AtomicInteger()
+    Async.blocking:
```

Comment:
General note on `Async.blocking` in tests: we heavily rely on the fact that returning `scala.concurrent.Future[T]` from MUnit tests will mean MUnit will try to run them to the end. Need to double check to make sure every test is doing this, or do something else (perhaps we can assume top-level `await` in test environment?)